### PR TITLE
Map Appsetting to flag names.

### PR DESCRIPTION
**Pull request by arijitAD**
This PR maps Golang flags to `Appsetting` struct.
For supporting configs for multiple ports, I was wondering to have `[]Appsetting` and have config name inside it that identifies the config and flags will be updated for that config only. There is always a default config. If the config name is not supplied then flags are updated in default config.
I was initially trying to support multiple configs in single struct but couldn't come up with an elegant approach. I am open to comments. 
Even though the current approach will have redundant values I chose it for simplicity.

### CODE REVIEW

File: elasticsearch.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 type ESUriErorr struct{}
 
 func (e *ESUriErorr) Error() string {
-	return "Wrong ElasticSearch URL format. Expected to be: scheme://host/index_name"
+	return "Wrong ElasticSearch URL format. Expected to be: scheme://Host/index_name"
```
Comment (slimus) [inline elasticsearch.go:18]: host

File: elasticsearch.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ type ESRequestResponse struct {
 
 // Parse ElasticSearch URI
 //
-// Proper format is: scheme://[userinfo@]host/index_name
+// Proper format is: scheme://[userinfo@]Host/index_name
```
Comment (slimus) [inline elasticsearch.go:58]: host

File: elasticsearch_test.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ func assertNoError(returnedError error, t *testing.T) {
 	}
 }
 
-// Argument host:port/index_name
+// Argument Host:port/index_name
```
Comment (slimus) [inline elasticsearch_test.go:35]: everywhere in comments and unexported structs we should revert to `host`

File: gor.go [hunk 4]
```diff
@@ -41,7 +41,7 @@ func main() {
 	// defer func() {
 	// 	if r := recover(); r != nil {
 	// 		fmt.Printf("PANIC: pkg: %v %s \n", r, debug.Stack())
-	// 	}
+	// 	}̦
```
Comment (slimus) [inline gor.go:44]: revert dot

File: http_client_test.go [hunk 5]
```diff
@@ -484,7 +484,7 @@ func TestHTTPClientErrors(t *testing.T) {
 	client = NewHTTPClient("http://not.existing", &HTTPClientConfig{Debug: true})
 	if resp, err := client.Send(req); err != nil {
 		if s := proto.Status(resp); !bytes.Equal(s, []byte("521")) {
-			t.Error("Should return status 521 for no such host, instead:", string(s))
+			t.Error("Should return status 521 for no such Host, instead:", string(s))
```
Comment (slimus) [inline http_client_test.go:487]: host

File: kafka.go [hunk 6]
```diff
@@ -10,12 +10,21 @@ import (
 
 // KafkaConfig should contains required information to
```
Comment (slimus) [inline kafka.go:11]: fix comment for exported struct

File: output_binary.go [hunk 7]
```diff
@@ -8,11 +8,11 @@ import (
...
 // BinaryOutputConfig struct for holding binary output configuration
 type BinaryOutputConfig struct {
-	workers        int
-	Timeout        time.Duration
-	BufferSize     int
-	Debug          bool
-	TrackResponses bool
+	workers        int           `json:"output-binary-workers"`
+	Timeout        time.Duration `json:"output-binary-timeout"`
+	BufferSize     int           `json:"output-tcp-response-buffer"`
```
Comment (slimus) [inline output_binary.go:13]:
```suggestion
	BufferSize     int           `json:"output-binary-tcp-response-buffer-size"`
```

File: output_http.go [hunk 8]
```diff
@@ -62,28 +62,28 @@ type response struct {
...
+	queueLen   int `json:"output-http-queue-len"`
 
-	elasticSearch string
+	elasticSearch string `json:"output-http-elasticsearch"`
 
-	Timeout      time.Duration
-	OriginalHost bool
-	BufferSize   int
+	Timeout      time.Duration `json:"output-http-timeout"`
+	OriginalHost bool          `json:"http-original-Host"`
```
Comment (slimus) [inline output_http.go:77]:
```suggestion
	OriginalHost bool          `json:"output-http-original-host"`
```

File: output_http.go [hunk 9]
```diff
@@ -62,28 +62,28 @@ type response struct {
 
 // HTTPOutputConfig struct for holding http output configuration
 type HTTPOutputConfig struct {
-	redirectLimit int
+	redirectLimit int `json:"output-http-redirects"`
```
Comment (slimus) [inline output_http.go:65]:
```suggestion
	redirectLimit int `json:"output-http-redirect-limit"`
```

File: output_http_test.go [hunk 10]
```diff
@@ -85,16 +85,16 @@ func TestHTTPOutputKeepOriginalHost(t *testing.T) {
 	input := NewTestInput()
 
 	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		if req.Host != "custom-host.com" {
+		if req.Host != "custom-Host.com" {
```
Comment (slimus) [inline output_http_test.go:88]: host

File: output_http_test.go [hunk 11]
```diff
@@ -85,16 +85,16 @@ func TestHTTPOutputKeepOriginalHost(t *testing.T) {
...
 			t.Error("Wrong header", req.Host)
 		}
 
 		wg.Done()
 	}))
 	defer server.Close()
 
-	headers := HTTPHeaders{HTTPHeader{"Host", "custom-host.com"}}
-	Settings.modifierConfig = HTTPModifierConfig{headers: headers}
+	headers := HTTPHeaders{HTTPHeader{"Host", "custom-Host.com"}}
```
Comment (slimus) [inline output_http_test.go:96]: host

File: settings.go [hunk 12]
```diff
@@ -29,64 +29,64 @@ func (h *MultiOption) Set(value string) error {
...
+	RecognizeTCPSessions bool   `json:"recognize-tcp-sessions"`
+	Pprof                string `json:"http-pprof"`
+
+	InputDummy   MultiOption `json:"input-dummy"`
+	OutputDummy  MultiOption `json:"output-dummy"`
+	OutputStdout bool        `json:"output-stdout"`
+	OutputNull   bool        `json:"output-null"`
+
+	InputTCP        MultiOption     `json:"input-tcp"`
+	InputTCPConfig  TCPInputConfig  `json:"input-tcp"`
```
Comment (slimus) [inline settings.go:47]: two keys have input-tcp json key, is it correct?

File: output_binary.go (hunk 7 above)
Comment (arijitAD) [reply output_binary.go:13]: I think the current name is fine, suggested name is too long for flag value.

File: settings.go (hunk 12 above)
Comment (arijitAD) [reply settings.go]: Fixed it.

**Comment by arijitAD** [comment]
@slimus Changed the settings to exported because in the next PR I will be reading them from a JSON file and directly parse it.
@urbanishimwe @slimus Addressed the comments.

**Pull request merged**